include/tvm/tir/transform.h

```diff
@@ -406,6 +406,14 @@ TVM_DLL Pass ConvertBlocksToOpaque();
  */
 TVM_DLL Pass CompactBufferAllocation();
 
+/*!
+ * \brief Flatten the multi-dimensional BufferLoad and BufferStore
+ *        to single dimensional Load/Store. Also remove Block to
+ *        ensure that the flattened TIR can not be scheduled again.
+ * \return The pass.
+ */
+TVM_DLL Pass FlattenBuffer();
```

Comment:
I wonder which name works better, `BufferFlatten` or `FlattenBuffer`, given that we are using `StorageFlatten` and `StorageRewrite` as other pass names

Reply:
There are mixed names, e.g. Simplify and MakePackedAPI are in different style. Let us go with this name for now